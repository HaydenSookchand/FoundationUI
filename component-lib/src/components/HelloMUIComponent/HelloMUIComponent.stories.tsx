import React from "react";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import HelloMUIComponent from "./HelloMUIComponent"; // Adjust path accordingly

export default {
  title: "An Example/HelloMUIComponent",
  component:HelloMUIComponent,
  parameters: {
    docs: {
      description: {
        component: "A simple MUI Typography component inside a Storybook story.",
      },
    },
  },
};

const theme = createTheme();

const Template = () => (
  <ThemeProvider theme={theme}>
    <HelloMUIComponent />
  </ThemeProvider>
);

export const Default = Template.bind({});
