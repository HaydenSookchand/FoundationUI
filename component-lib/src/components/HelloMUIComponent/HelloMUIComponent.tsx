import React from "react";
import { Typography, Box, styled } from "@mui/material";
import theme from '../../theme/theme';
import colors from '../../theme/colors';

const StyledContainer = styled(Box)(() => ({
  padding: theme.spacing.xl,
  backgroundColor: theme.palette.background.default,
  borderRadius: theme.spacing.lg,
  border: `2px solid ${colors.outlinedBorder}`,
}));

const HelloMUIComponent = () => {
  return (
    <StyledContainer>
      <Typography variant="h4" sx={{ color:theme.palette.text.primary }}>
        Hello World MUI JSX
      </Typography>
    </StyledContainer>
  );
};

export default HelloMUIComponent;