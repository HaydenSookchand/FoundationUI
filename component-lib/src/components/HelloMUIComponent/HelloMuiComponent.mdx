
import { Meta, Primary, Controls, Story } from '@storybook/blocks';
import HelloMUIComponent from './HelloMUIComponent';
import * as HelloMUIComponentStories from './HelloMUIComponent.stories';

<Meta title="HelloMUIComponent" component={HelloMUIComponent} of={HelloMUIComponentStories} />

# Hello MUI Button

<Story of={HelloMUIComponentStories.Default} />
<br /> <br />


## Design Principle

Talk about the REASON for this component

<br />

## Usage

Use the  component to initiate actions such as ...

### Example

Hereâ€™s an example of using the component in your application:

```jsx
   <HelloMUIComponent />
```


## Props
 
<Controls />

## Variants 
