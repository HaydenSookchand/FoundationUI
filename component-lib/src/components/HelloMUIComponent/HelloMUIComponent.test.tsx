import React from "react";
import { screen, renderWithTheme } from "../../tests/test-utils"; // âœ… Use shared test utility
import "@testing-library/jest-dom";
import HelloMUIComponent from "./HelloMUIComponent";

describe("HelloMUIComponent", () => {
  it("renders the component with correct text", () => {
    renderWithTheme(<HelloMUIComponent />);
    expect(screen.getByText("Hello World MUI JSX")).toBeInTheDocument();
  });

  it("applies correct typography color", () => {
    renderWithTheme(<HelloMUIComponent />);
    const textElement = screen.getByText("Hello World MUI JSX");
    expect(textElement).toHaveStyle({ color: "#FFB900" }); // Uses mock theme
  });

  it("matches snapshot", () => {
    const { asFragment } = renderWithTheme(<HelloMUIComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
