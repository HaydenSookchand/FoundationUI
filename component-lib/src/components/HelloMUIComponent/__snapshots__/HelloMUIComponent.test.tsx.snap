// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloMUIComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-xp6dn9"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-1c0vpmj-MuiTypography-root"
    >
      Hello World MUI JSX
    </h4>
  </div>
</DocumentFragment>
`;
